File: PolyCheck.cpp
// PolyCheck.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include <iostream>
#include <fstream>
#include <sstream>
#include <map>
using namespace std;

struct Point {

	int x, y;

	Point() {
		x = 0;
		y = 0;
	}

	Point(int a, int b) {
		x = a;
		y = b;
	} // constructor

	~Point() {}

}; // Point

/**************************************************************************/

int sizeCheck(string name) {
	cout << "Initial readthrough:" << endl;
	int i = 0;
	fstream f(name);
	int x, y;
	while (f >> x) {
		f >> y;
		cout << "(" << x << "," << y << ")" << endl;
		i++;
	}
	f.close();
	return i;
}

int check(Point p[], int size) {

	int code = 200;
	if (size <= 2) return 400;

	for (int i = 1; i < size; i++) {
		for (int j = 0; j < i; j++)
		if (p[i].x == p[j].x && p[i].y == p[j].y) {
			cout << " Problem: (" << p[i].x << "," << p[i].y << ")" << endl;
			code = 401;
		}
	}
	return code;
} // check

int main(int argc, char* argv[])
{
	if (argc < 1 || argv[1] == NULL) {
		cout << "Run as:\n PolyCheck file.txt" << endl;
		return 0;
	}
	string args = argv[1];
	int size = sizeCheck(args);
	Point* p = new Point[size];
	int index = 0;

	fstream f(args);
	int x, y;
	for (int i = 0; i < size; i++) {
		f >> x;
		f >> y;
		p[i] = Point(x, y);
	}

	int ok = check(p,size);
	if (ok == 200)
		cout << "OK " << ok << ": Polygon is valid." << endl;
	else if (ok == 400)
		cout << "ERR " << ok << ": Polygon features two or fewer data points." << endl;
	else if (ok == 401)
		cout << "ERR " << ok << ": Duplicate points detected." << endl;


	delete[] p;
	f.close();
	return 0;
}

File: PolyCheck.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.*;

public class PolyCheck {

	public static void main(String[] args) throws FileNotFoundException {

		if (args.length == 0) {
			System.out.println("Run as:\n java PolyCheck file.txt");
			return;
		}

		int size = sizeCheck(args[0]);
		Point[] p = new Point[size];

		File f = new File(args[0]);
		Scanner s = new Scanner(f);
		int x, y;
		for (int i = 0; i < p.length; i++) {
			x = s.nextInt();
			y = s.nextInt();
			p[i] = new Point(x, y);
		}

		int ok = check(p);
		if (ok == 200)
			System.out.println("OK " + ok + ": Polygon is valid.");
		else if (ok == 400)
			System.out.println("ERR" + ok + ": Polygon features two or fewer data points.");
		else if (ok == 401)
			System.out.println("ERR" + ok + ": Duplicate points detected.");

	} // main

	public static int sizeCheck(String name) throws FileNotFoundException {
		int i = 0;

		System.out.println("Initial readthrough:");
		File f = new File(name);
		Scanner s = new Scanner(f);
		int x, y;
		while (s.hasNextInt()) {
			x = s.nextInt();
			y = s.nextInt();
			System.out.println("(" + x + "," + y + ")");
			i++;
		}
		s.close();
		return i;
	}

	public static int check(Point[] p) {

		int code = 200;
		if (p.length <= 2)
			return 400;

		for (int i = 1; i < p.length; i++) {
			for (int j = 0; j < i; j++)
				if (p[i].x == p[j].x && p[i].y == p[j].y) {
					System.out.println(" Problem: (" + p[i].x + "," + p[i].y + ")");
					code = 401;
				}
		}
		return code;

	} // check

} // PolyCheck
